### PR TITLE
Fixed what caused #7315

### PR DESCRIPTION
**Pull request by kazarmy**
Test at radare/radare2-regressions#816. Supersedes #7331.

### CODE REVIEW

File: libr/parse/parse.c [hunk 1]
```diff
@@ -321,7 +321,10 @@ static int filter(RParse *p, RFlag *f, char *data, char *str, int len, bool big_
 			case 2: // hack for ascii
 				tmp_count = 0;
 				for (tmp = data; tmp < ptr; tmp++) {
-					if (*tmp == '[') {
+					if (*tmp == 0x1b) {
+						while (tmp < ptr-1 && *tmp != 'm') tmp++;
```
Comment (radare) [inline libr/parse/parse.c]: Add {} braces

**Comment by radare** [comment]
Are you sure this patch works without colors enabled too? Can we have some tests in r2r to verify any brokenness? Because right now its green

File: libr/parse/parse.c (hunk 1 above)
Comment (radare) [inline libr/parse/parse.c]: Add spaces around -

Comment (kazarmy) [inline libr/parse/parse.c]: Ok done.

**Comment by kazarmy** [comment]
Yes and yes. Just added a test to radare/radare2-regressions#816.

**Comment by kazarmy** [comment]
Changed test string in radare/radare2-regressions#816 to `m0[<ESC>` (hexpairs 1b5b306d). Works fine.

**Pull request merged**